pyzoo/zoo/zouwu/model/anomaly.py

```diff
@@ -126,15 +140,21 @@ def detect(self,
                         threshold.shape), "is not valid")
         elif isinstance(threshold, tuple) \
                 and len(threshold) == 2 \
-                and threshold.shape == y.shape:
+                and threshold[0].shape == y.shape[1:] \
+                and threshold[-1].shape == y.shape[1:]:
             return self._check_range(y, threshold)
         else:
             raise ValueError(
                 "threshold shape", str(threshold),
                 "is not valid")
 
-    def _check_all_distance(self, y, yhat, threshold):
-        pass
+    def _check_all_distance(self, y, yhat):
+        index = []
+        for i in range(y.shape[0]):
+            diff = self.dist_measure.distance(y[i], yhat[i])
+            if diff >= self.threshold:
+                index.append(i)
+        return index
 
     def _check_per_dim_distance(self, y, yhat, threshold):
         pass
```

Comment:
these didn't implement, you may just raise Error. pass it silently may not be proper.